### ISSUE TITLE
Node_modules get cleared with package name missing error in the custom plugin for Reaciton Api.

### ISSUE THREAD

**Issue by ankurpata** [reactioncommerce/reaction#6246]
Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues] 

## Issue Description
When tried to add a new dependency on to plugin in the package.json, the ```make init-dev```command on reaction  root directory should install all the dependencies provided in the custom plugin. Instead it clear the node_modules and fails to install again.
Screenshots of node_modules and package.json in the custom plugin:
![image](https://user-images.githubusercontent.com/3061592/81749007-e8938a80-94bb-11ea-8750-0b495c9febcc.png)
![image](https://user-images.githubusercontent.com/3061592/81749061-0103a500-94bc-11ea-9768-b8a968caad1e.png)

Error on Reaction api Docker logs:
![image](https://user-images.githubusercontent.com/3061592/81749249-5344c600-94bc-11ea-88ec-9a2f8da08985.png)

## Steps to Reproduce
1.  Create a custom plugin and integrate it with Reaction Api.
2.  Add dev dependencies like ```node-rdkafka```
3. Try running `make init-dev` 
4. Fails installing custom plugin.

## Possible Solution
I faced this issue on MacOs, the docker must be using node-gyp to install dependencies for any additional npm package. It might work if it installed using ubuntu.

### Versions
3.7.0

**Comment by aldeed** [comment]
@ankurpata It would be helpful if you can link to a repo with a simple reproduction of this. There could be a lot of things happening, and we'd need to see exactly how you're linking the plugin and exactly how it is listing its dependencies. 

I notice you say `Add dev dependencies`. If that means that `node-rdkafka` is listed in `devDependencies` instead of `dependencies` then that's probably the issue. Only `dependencies` would be installed because that's how NPM works.

**Comment by loan-laux** [comment]
@aldeed I'm having the same problem and I've created a flattened reproduction repo [right here](https://github.com/loan-laux/repro-npm-rc/tree/master). Clone it, run `make init-dev-reaction` and you should run into the same issue as @ankurpata reported.

Also, I confirm that it's happening with production dependencies. If you run `npm install` in the `@reactioncommerce/*` directories from the host machine before starting `reaction`, their `node_modules` get wiped except for `fsevents`. Then Reaction throws an error about not being able to solve the packages' own deps (of course).

**Comment by loan-laux** [comment]
On another note, since this looks reproducible and prevents (some) developers from working on mounted NPM modules, I would suggest that it deserves at least a `severity medium` tag.

**Comment by aldeed** [comment]
@loan-laux @ankurpata  Oh I see! I did not realize this was referring to plugins linked through volume mounts. In that case, yes we're aware of this issue. There is really no solution we could find when using volume mounts, which is why I created a different solution for local plugin development here: https://github.com/reactioncommerce/reaction/pull/6238

Using that branch, you should be able to do `make dev-reaction` _first_ and _then_ copy in each local plugin using the new script while the API is running. That PR should be good to merge into `trunk`, but I wanted a few people to try it out and confirm it's working for more than just me.

I'm hoping to document this and other aspects of local development that have changed this week.

**Comment by loan-laux** [comment]
@aldeed This works great and looks exactly like what we needed. Thank you so much! @ankurpata, if this works for you, please close this issue.

**Comment by ankurpata** [comment]
Cool, I'll try this out and close this if it works. 
Thanks @aldeed, @loan-laux

**Comment by nnnnat** [comment]
@ankurpata please re-open if this is still an issue for you.
